pootle/apps/pootle_app/models/directory.py

```diff
@@ -154,15 +154,17 @@ def can_be_updated(self):
     def get_children(self):
         result = []
         if not self.is_projects_root():
-            # FIXME: can we replace this with a quicker path query?
-            result.extend([item for item in
-                           self.child_stores.live().iterator()])
-            result.extend([item for item in self.child_dirs.live().iterator()])
+            result.extend(
+                self.child_stores.live().select_related(
+                    "parent", "translation_project"))
+            result.extend(self.child_dirs.live())
```

Comment:
dont know atm - but i do know it gets called in the stats code somewhere

personally i think we should rethink the stats code so it calculates the list of objects to expire only once